### PR TITLE
rewrite makefile & docker-compose

### PR DESCRIPTION
**Pull request by mykiwi**
```
$ make
start:           Install and start the project
stop:            Remove docker containers
reset:           Reset the whole project
db:              Reset the database and load fixtures
:---------------------------------------------------------------------------:
new-migration:   Generate a migration by comparing your current database to your mapping information.
watch:           Watch the assets and build their development version on change
clear:           Remove all the cache, the logs, the sessions and the built assets
clean:           Clear and remove dependencies
:---------------------------------------------------------------------------:
test:            Run the PHP and the Javascript tests
tu:              Run the PHP unit tests
tf:              Run the PHP functional tests
tj:              Run the Javascript tests
lint:            Run lint on twig, yaml and Javascript files
ls:              Lint Symfony (twig and yaml) files
lj:              Lint the Javascript to follow the convention
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -15,39 +15,10 @@ services:
...
-            MYSQL_PASSWORD: enmarche_test
+            MYSQL_ROOT_PASSWORD: root
 
     redis:
         image: redis:3.2
 
     rabbitmq:
         image: rabbitmq:3-management
-
-    tools:
```
Comment (tgalopin) [inline docker-compose.yml]: That's really great to finally remove the tools container. +1 :)

File: docker-compose.yml [hunk 2]
```diff
@@ -15,39 +15,10 @@ services:
...
-        build:
-            context: .
-        command: sleep infinity
-        depends_on:
-            - db
-        volumes_from:
-            - app
-
-    pma:
```
Comment (tgalopin) [inline docker-compose.yml]: Nice too IMO :)

File: Makefile [hunk 3]
```diff
@@ -1,88 +1,112 @@
...
+clear: perm
+	-$(EXEC) rm -rf var/cache/*
+	-$(EXEC) rm -rf var/sessions/*
+	-$(EXEC) rm -rf supervisord.log supervisord.pid npm-debug.log .tmp
+	rm -rf var/logs/*
+	rm -rf web/built
 
-stop:           ## Stop the Docker containers
-	docker-compose down
+clean:          ## Clear and remove dependencies
```
Comment (tgalopin) [inline Makefile]: Shouldn't `clean` include `clear` ?

File: Makefile [hunk 4]
```diff
@@ -1,88 +1,112 @@
...
 
-clear-all:      ## Deeply clean the application (remove all the cache, the logs, the sessions and the built assets)
-	$(CONSOLE) cache:clear --no-warmup
-	$(CONSOLE) cache:clear --no-warmup --env=prod
-	$(CONSOLE) cache:clear --no-warmup --env=test
-	$(TOOLS) rm -rf var/logs/*
-	$(TOOLS) rm -rf var/sessions/*
-	$(TOOLS) rm -rf web/built
-	$(TOOLS) rm -rf supervisord.log supervisord.pid npm-debug.log .tmp
+fig-build:
```
Comment (tgalopin) [inline Makefile]: What about simply `build`?

File: Makefile [hunk 5]
```diff
@@ -1,88 +1,112 @@
...
+	$(FIG) build
 
-clean:          ## Removes all generated files
-	- @make clear-all
-	$(TOOLS) rm -rf vendor node_modules
+fig-up:
```
Comment (tgalopin) [inline Makefile]: What about simply `up`?

File: docker-compose.override.yml.dist [hunk 6]
```diff
@@ -16,14 +19,16 @@ services:
             - "3306:3306"
 
     pma:
+        image: phpmyadmin/phpmyadmin
+        depends_on:
+            - db
+        environment:
+            PMA_HOST: db
+            PMA_USER: enmarche
```
Comment (tgalopin) [inline docker-compose.override.yml.dist]: Isn't it `root/root` now?

File: Makefile (hunk 3 above)
Comment (mykiwi) [inline Makefile]: It does, line 49

File: docker-compose.override.yml.dist (hunk 6 above)
Comment (mykiwi) [inline docker-compose.override.yml.dist]: good catch

File: Makefile [hunk 7]
```diff
@@ -1,88 +1,112 @@
...
+	$(RUN) vendor/bin/phpunit --exclude-group functionnal
 
-db-create:      ## Create the database
-	$(TOOLS) php -r "for(;;){if(@fsockopen('db',3306)){break;}}" # Wait for MariaDB
-	$(CONSOLE) doctrine:database:drop --force --if-exists
-	$(CONSOLE) doctrine:database:create --if-not-exists
-	$(CONSOLE) doctrine:migrations:migrate -n
+tf:             ## Run the PHP functional tests
+tf: vendor
+	$(RUN) vendor/bin/phpunit --group functionnal
```
Comment (hhamon) [inline Makefile]: the functional group is now called `functional`. I fixed the typo in all tests already.

Comment (hhamon) [inline Makefile]: Forget it, it's not true for now as my PR is not merged yet.

File: Makefile [hunk 8]
```diff
@@ -1,88 +1,112 @@
...
+
+stop:           ## Remove docker containers
+	$(FIG) kill
+	$(FIG) rm -v --force
+
+reset:          ## Reset the whole project
+reset: stop start
+
+db:             ## Reset the database and load fixtures
+db: vendor
```
Comment (mykiwi) [inline Makefile]: Because `bin/console` needs `vendor`

**Comment by rpg600** [comment]
Can we merge this pr ?

**Pull request merged**